index.d.ts

```diff
@@ -0,0 +1,81 @@
+export interface Action {
+  type: string;
+}
+
+
+/* reducers */
+
+export type Reducer<S> = <A extends Action>(state: S, action: A) => S;
+
+export function combineReducers<S>(reducers: {[key: string]: Reducer<any>}): Reducer<S>;
```

Comment:
I think that reducers map should be defined separately:

```typescript
export interface ReducersMapObject {
    [key: string]: Reducer<any>;
}

export function combineReducers<S, M extends ReducersMapObject>(reducers: M): Reducer<S>;
```

So anyone can easily make his own type for this corresponding with his `State` type if one was defined:

```typescript
interface MyReducers extends ReducersMapObject {
    'posts': Reducer<Post[]>;
    'user': Reducer<MyUser>;
    … 
}

const reducers: MyReducers = …;

const myRootReducer = combineReducers<Reducer<State>, MyReducers>(reducers);
```